CODEOWNERS

```diff
@@ -1,4 +1,52 @@
 # Users referenced in this file will automatically be requested as reviewers for PRs that modify the given paths.
 # See https://help.github.com/articles/about-code-owners/
 
-* @dotnet/dotnet-docker-reviewers
+### General infra ###
+/* @dotnet/dotnet-docker-reviewers
+eng/ @dotnet/dotnet-docker-reviewers
+
+### Dockerfiles ###
+
+# common paths
+src/**/helix/ @dotnet/dnceng
+src/**/cross*/ @dotnet/runtime-infrastructure
+src/**/webassembly*/ @dotnet/runtime-infrastructure
+
+# almalinux
+src/almalinux/**/source-build/ @dotnet/source-build-internal
+
+# alpine
+src/alpine/**/amd64/ @dotnet/dotnet-source-build-internal
+
+# azurelinux
+src/azurelinux/**/android/ @dotnet/runtime-infrastructure
+src/azurelinux/**/fpm/ @dotnet/runtime-infrastructure
+src/azurelinux/**/opt/ @dotnet/runtime-infrastructure
+
+# cbl-mariner
+src/cbl-mariner/**/android/ @dotnet/runtime-infrastructure
+src/cbl-mariner/**/fpm/ @dotnet/runtime-infrastructure
+src/cbl-mariner/**/opt/ @dotnet/runtime-infrastructure
+src/cbl-mariner/**/docker-testrunner/ @dotnet/dotnet-docker-reviewers
+src/cbl-mariner/2.0/amd64/ @dotnet/runtime-infrastructure
+
+# centos
+src/centos/stream9/amd64/ @dotnet/source-build-internal
+
+# debian
+src/debian/11/amd64/ @dotnet/source-build-internal
+src/debian/11/arm64v8/ @dotnet/source-build-internal

```

Comment:
FYI I am removing this in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1158.